docs/configuration.md

```diff
@@ -0,0 +1,64 @@
+# Configuration
+
+Here is the list of available properties:
+
+| **Property Name**               | **Default Value** | **Explanation**                                                                                                                                                                             |
+|---------------------------------|-------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `shutdownTimeoutMs`             | 90,000ms          | Maximum number of milliseconds a resource shutting down via `BaseReactiveResourceShutdownStrategy` should take to shutdown.                                                                 |
+| `clientsReactionTimeMs`         | 30,000ms          | How many milliseconds to wait after calling `prepareForShutdown` on each `GracefulShutdownStrategy` so that all clients understand that they should not call this application node anymore. |
+| `strategiesCheckIntervalTimeMs` | 5,000ms           | How long to wait in milliseconds between checking the `canShutdown` flag of each `GracefulShutdownStrategy` being executed.                                                                 |
+| `resourceCheckIntervalTimeMs`   | 250ms             | How long to wait in milliseconds between checking termination status of a resource shutting down via the `BaseReactiveResourceShutdownStrategy`.                                            |
+
+To configure the properties, edit your `application-*.yml` file in your `resources` folder as such:
+```yaml
+tw-graceful-shutdown:
+  shutdown-timeout-ms: <INTEGER>
+  client-reaction-time-ms: <INTEGER>
+  strategies-check-interval-time-ms: <INTEGER>
+  resource-check-internal-time-ms: <INTEGER>
+```
+
+### Health Indicator
+```yaml
+tw-graceful-shutdown:
+  health-indicator:
+    enabled: [true|default:false]
+```
+If set to `true`, creates `GracefulShutdownHealthStrategy` and `GracefulShutdownHealthIndicator` beans conditionally on the presence of the `org.springframework.boot.actuate.health.AbstractHealthIndicator` class.
+
+### Kagkarlsson DB Scheduler
+```yaml
+tw-graceful-shutdown:
+  kagkarlsson-db-scheduler:
+    enabled: [true|default:false]
+```
+If set to `true`, creates the `KagkarlssonDbScheduledTaskShutdownStrategy` bean conditionally on the presence of the `com.github.kagkarlsson.scheduler.Scheduler` class and bean. Used to gracefully shutdown tasks that are being executed by the scheduler.
+
+### Spring Task Scheduler
+```yaml
+tw-graceful-shutdown:
+  spring-task-scheduler:
+    enabled: [true|default:false]
```

Comment:
Is default value - false? Because by default this property is not present. And we don't have application.yml in this project. That means that this and similar properties is missing. In that case startegies by default have this configuration `@ConditionalOnProperty(value = "tw-graceful-shutdown.kagkarlsson-db-scheduler.enabled", matchIfMissing = true)` so that means, that if property is not provided, then it is enabled. Maybe default value in that case should be - true.